sqlite_query.c

```diff
@@ -423,44 +426,101 @@ sqlite_convert_to_pg(Form_pg_attribute att, sqlite3_value * val, AttInMetadata *
 
 							buffer = int642binstr(sqlti, buffer, SQLITE_FDW_BIT_DATATYPE_BUF_SIZE);
-							elog(DEBUG4, "sqlite_fdw : BIT buf l=%ld v = %s", SQLITE_FDW_BIT_DATATYPE_BUF_SIZE, buffer);
 							valstr = buffer;
-							break; /* !!! use valstr later! */
+							elog(DEBUG4, "sqlite_fdw : BIT buf l=%ld v = %s", SQLITE_FDW_BIT_DATATYPE_BUF_SIZE, buffer);
+							break;
 						}
 					case SQLITE_FLOAT:
 					case SQLITE_BLOB:
+					default:
 						{
 							sqlite_value_to_pg_error();
 							break;
 						}
 					case SQLITE3_TEXT:
-						{
-							if (value_byte_size_blob_or_utf8)
+					{
+						if (value_byte_size_blob_or_utf8)
+							sqlite_value_to_pg_error();
+						else
+							pg_column_void_text_error();
+						break;
+					}
+				}
+				break;
+			}
+		case BPCHAROID:
+		case VARCHAROID:
+		case CHAROID:
+		case TEXTOID:
+		case JSONOID:
+		case JSONBOID:
+		case NAMEOID:
+		case DATEOID:
+		case TIMEOID:
+			{
+				valstr = sqlite_text_value_to_pg_db_encoding(val);
+				break;
+			}
+		default:
+			{
+				/* PostGIS data types can be supported only by name
+				 * This is very rare and not fast algorythm branch
+				 */
+				char *pg_dataTypeName = TypeNameToString(makeTypeNameFromOid(att->atttypid, att->atttypmod));
+				NameData	pgColND = att->attname;
+
+				if (listed_datatype(pg_dataTypeName, postGisSpecificTypes))
+				{
+					elog(DEBUG4, "sqlite_fdw : is postGisSpecificType %s", pg_dataTypeName);
+					ereport(ERROR, (errcode(ERRCODE_FDW_INVALID_DATA_TYPE),
+									errmsg("This data type is PostGIS specific and is not supported by SpatiaLite"),
+									errhint("Data type: \"%s\" in column \"%.*s\"", pg_dataTypeName, (int)sizeof(pgColND.data), pgColND.data)));
+				}
+
+				if (listed_datatype(pg_dataTypeName, postGisSQLiteCompatibleTypes))
+				{
+					elog(DEBUG4, "sqlite_fdw : is postGisSQLiteCompatibleType");
+					switch (sqlite_value_affinity)
+					{
+						case SQLITE_INTEGER:
+						case SQLITE_FLOAT:
+						default:
+							{
 								sqlite_value_to_pg_error();
-							else
-								pg_column_void_text_error();
+								break;
+							}
+						case SQLITE_BLOB: /* <-- proper and recommended SQLite affinity of value for pgtyp */
+						{
+							#ifdef SQLITE_FDW_GIS_ENABLE
+							unsigned const char * p_blob = sqlite3_value_blob(val);
+							valstr = SpatiaLiteAsPostGISgeom (
+								(struct blobOutput){(const char *)p_blob, value_byte_size_blob_or_utf8},
+								att
+							);
+							/* !!! use valstr later! Hex Input */
+							#else
+							ereport(ERROR, (errcode(ERRCODE_FDW_INVALID_DATA_TYPE),
+											errmsg("This PostGIS data type is supported by SpatiaLite, but FDW compiled without GIS data support"),
+											errhint("Data type: \"%s\" in column \"%.*s\"", pg_dataTypeName, (int)sizeof(pgColND.data), pgColND.data)));
+							#endif
 							break;
 						}
-					default:
+						case SQLITE3_TEXT:
 						{
+							/* WKT data transport between PostGIS and SpatiaLite ? */
```

Comment:
Because this is a comment for an incomplete part, I cannot accept it into master branch.  
Please move it to your internal branch and add it when you are ready about that WKT processing.

Reply:
What about such variant as https://github.com/pgspider/sqlite_fdw/pull/96/commits/ce5574f79596fbebd4ef0f7a72eba935a8148a1d , @son-phamngoc  ?

Reply:
Okay. Please help me to resolve this comment.